phoenicis-engines/src/main/java/org/phoenicis/engines/EnginesSource.java

```diff
@@ -45,17 +46,22 @@ public EnginesSource(ScriptInterpreter scriptInterpreter, ObjectMapper objectMap
         this.objectMapper = objectMapper;
     }
 
-    public void fetchAvailableEngines(Consumer<List<EngineCategoryDTO>> callback) {
+    public void fetchAvailableEngines(List<CategoryDTO> categoryDTOS, Consumer<List<EngineCategoryDTO>> callback) {
         final InteractiveScriptSession interactiveScriptSession = scriptInterpreter.createInteractiveSession();
 
         List<EngineCategoryDTO> engines = new ArrayList<>();
-        interactiveScriptSession.eval("include([\"Engines\", \"Wine\", \"Engine\", \"Object\"]);",
-                ignored -> interactiveScriptSession.eval("new Wine().getAvailableVersions()", output -> {
-                    EngineCategoryDTO wine = new EngineCategoryDTO.Builder().withName("Wine").withDescription("Wine")
-                            .withSubCategories(unSerialize(output)).build();
-                    engines.add(wine);
-                    callback.accept(engines);
-                }, this::throwError), this::throwError);
+        for (CategoryDTO categoryDTO : categoryDTOS) {
+            final String engineName = categoryDTO.getName();
+            interactiveScriptSession.eval("include([\"Engines\", \"" + engineName + "\", \"Engine\", \"Object\"]);",
+                    ignored -> interactiveScriptSession.eval("new " + engineName + "().getAvailableVersions()",
+                            output -> {
+                                EngineCategoryDTO wine = new EngineCategoryDTO.Builder().withName(engineName)
+                                        .withDescription(engineName).withSubCategories(unSerialize(output)).build();
+                                engines.add(wine);
+                                callback.accept(engines);
```

Comment:
Mh. Just tried. It's not working for some reason. The `engines` list is empty at that point.